pyttb/sptensor.py

```diff
@@ -440,43 +550,60 @@ def collapse(
                     self.subs[:, remdims].transpose()[0],
                     self.vals.transpose()[0],
                     size=newsize[0],
-                    func=fun,
+                    func=function_handle,
                 )
             return np.zeros((newsize[0],))
 
         # Create Result
         if self.subs.size > 0:
             return ttb.sptensor.from_aggregator(
-                self.subs[:, remdims], self.vals, tuple(newsize), fun
+                self.subs[:, remdims], self.vals, tuple(newsize), function_handle
             )
         return ttb.sptensor(np.array([]), np.array([]), tuple(newsize), copy=False)
 
-    def contract(self, i: int, j: int) -> Union[np.ndarray, sptensor, ttb.tensor]:
+    def contract(self, i1: int, i2: int) -> Union[np.ndarray, sptensor, ttb.tensor]:
```

Comment:
I think snake case feels a bit more pythonic. For our example S1 etc doesn't matter too much but for an interface it just looks a little awkward to me. Also zero indexing?
```suggestion
    def contract(self, i_0: int, i_1: int) -> Union[np.ndarray, sptensor, ttb.tensor]:
```